cmd/main.go

```diff
@@ -239,22 +250,79 @@ func scanUniversalChain(ctx context.Context, c *config.Config, client scan.EthCl
 				waitBeforeNextScan(ctx, c.WaitingTime)
 				break
 			}
-			// discovering new contracts deployed on the ownership chain
-			// choosing which contracts to scan
-			// if contracts come from flag, consider only those that have been discovered (whether in this iteration or previously)
-			// load merkle trees for all contracts whose events have to be scanned for
-			// scanning contracts for events on the ownership chain
-			// getting transfer events from scan events
-			// retrieving minted events and update the state accordingly
-			nextStartingBlock, err := scanAndDigest(ctx, stateService, c, s, startingBlock, lastBlock, lastOwnershipBlockTimestamp, client)
+
+			err = processUniversalBlockRange(ctx, c, client, stateService, s, startingBlock, lastBlock, lastOwnershipBlockTimestamp)
 			if err != nil {
+				if errors.Is(err, ReorgError{}) {
+					return err
```

Comment:
shouldn't this be Fatal for now? at least until we support reorgs.

Reply:
No, upon reaching this point, the program will exit the for loop with an error, causing the node to shut down.